metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DatasetService.java

```diff
@@ -202,6 +204,21 @@ public void updateDataset(MetisUser metisUser, Dataset dataset, String xsltStrin
     datasetDao.update(dataset);
   }
 
+  private void cleanDatasetXslt(ObjectId xsltId) {
```

Comment:
I'm a bit worried about backwards compatibility. Is it your intention to do a database migration? If not, I have a concern. 

What if an xslt is used, but it is not referred to by ID from a transformation plugin, but by URL (because it is old). Then this method will conclude that the xslt is not referenced at all, and clean it up. With the loss of history as a consequence.